.github/workflows/terraform_gh_runner.yml

```diff
@@ -19,40 +21,47 @@ jobs:
       - name: Set up Terraform
         uses: hashicorp/setup-terraform@v1
         with:
-          terraform_version: "1.4.2"
+          terraform_version: "1.5.7"
           cli_config_credentials_token: ${{ secrets.TF_CLOUD_TOKEN }}
 
       - name: Install dependencies
         run: |
           # Install any dependencies required by your Terraform code
 
       - name: Run Bash Script
+        env:
+          GH_TOKEN: ${{ github.token }}
```

Comment:
You're using `secrets.GITHUB_TOKEN` above and `github.token` here, which are both the same thing. Would be nice to make it consistent.

But this environment doesn't seem to be used either.

Reply:
It is used, will make consistent. It is needed for api calls to get the PAT token for github runner registration.

Reply:
GitHub API again? Which API, I don't see any usage of it.

Reply:
Ok, for $token in the auth header of the curl request that is the pat token to authenticate to the repo and get the runner token. If i remove GH_TOKEN you will see an error in ci that you need to have GH_TOKEN to call github apis from workflow :). I wasn't using it before, but added it to fix the problem. Here is an old version:
https://github.com/subspace/infra/blob/bd296590f29cdd730ab29869c7ade1258cb395ae/.github/workflows/terraform_gh_runner.yml